scripts/build_ses.sh

```diff
@@ -0,0 +1,284 @@
+role-igw/stack/default/ceph/minions/${igw}.yml
+role-igw/cluster/${igw}.sls
+EOF
+    done
+    # Rados gateways
+    rgws=( $(rnd_sel_nodes $NUM_RGW $master ${minions[@]}) )
+    for rgw in ${rgws[@]}
+    do
+        run_ssh_cmd $master "echo 'role-rgw/cluster/${rgw}.sls' >> $pconf"
+    done
+
+    cat <<EOF | ssh $master "cat >> $pconf"
+# common configuration
+config/stack/default/global.yml
+config/stack/default/ceph/cluster.yml
+
+# profile assignment
+EOF
+
+    # Need to assign profiles, we'll just add whatever was found
+    for profile in $(ssh $master "find $(dirname $pconf) -name 'profile*'")
+    do
+        profile=$(echo $profile |sed "s#^$(dirname $pconf)/##g")
+        cat <<EOF | ssh $master "cat >> $pconf"
+$profile/cluster/*.sls
+$profile/stack/default/ceph/minions/*.yml
+EOF
+    done
+}
+
+function install_ses() {
+    echo "== run stage 0 (prep) =="
+    # Note: we don't use run_ssh_cmd because a reboot during prep returns a > 0 errorcode.
+    ssh $master "salt-run state.orch ceph.stage.prep"
+    prep_return=$?
+
+    # there is a chance the nodes will reboot so ping check them
+    if [ $prep_return -gt 0 ]
+    then
+        # wait a big for the nodes to shutdown.
+        sleep 10
+    fi
+    for hn in $master ${minions[@]}
+    do
+        echo "Attempting to ssh to $hn"
+        ip_good=0
+        for i in $(seq 30)
+        do
+            ssh $hn "zypper --gpg-auto-import-keys ref -f"
+            if [ $? -eq 0 ]
+            then
+                ip_good=1
+                # refresh repoistories after the reboot, as cloud repo can update key.
+                ssh $hn "zypper --gpg-auto-import-keys ref -f"
+                break
+            fi
+            sleep 2
+        done
+        if [ ip_good -eq 0 ]
+        then
+            echo "failed to contact $hn"
+            exit 3
+        fi
+    done
+    if [ $prep_return -gt 0 ]
+    then
+        # prep stage might have needed to reboot, so need to run it a second time
+        echo "== run stage 0 (prep) - take 2 =="
+        run_ssh_cmd $master "salt-run state.orch ceph.stage.prep"
+    fi
+
+    echo "== run stage 1 (discovery) ==" # - collect data from the nodes
+    run_ssh_cmd $master "salt-run state.orch ceph.stage.discovery"
+
+    echo "== generate policy.cfg ==" # from data found during discovery
+    generate_policy_config
+
+    echo "== run stage 2 (configure) =="
+    run_ssh_cmd $master "salt-run state.orch ceph.stage.configure"
+
+    echo "== Setting networks =="
+    if [ -n "$public_network" ]
+    then
+        echo "public_network: $public_network"
+        cat <<EOF | ssh $master "cat >> /srv/pillar/ceph/stack/ceph/cluster.yml"
+public_network: $public_network
+EOF
+        if [ -n "$cluster_network" ]
+        then
+            echo "cluster_network: $cluster_network"
+            cat <<EOF | ssh $master "cat >> /srv/pillar/ceph/stack/ceph/cluster.yml"
+cluster_network: $cluster_network
```

Comment:
This is providing an incorrect public network.
In my local setup it should have given 192.168.124.0/24 in cluster.yml
But it has given 192.168.122.0/24. (or similar off by 1 or 2)
Can you please double check that?

Reply:
Will do, thanks. 
also am making it call bash build_ses.sh so the executable bit doesn't get in the way. Nice fine and thanks for testing!

Reply:
So the 122 is coming from the makecloud net_public variable: https://github.com/matthewoliver/automation/blob/external_ceph/scripts/lib/mkcloud-common.sh#L362

I assume your overriding that value? or maybe it's failing to override in qa_crowbarsetup.sh when running onadmin_external_ceph. (https://github.com/matthewoliver/automation/blob/external_ceph/scripts/qa_crowbarsetup.sh#L5183)

Reply:
@kbaikov Is it still getting the wrong public network?

Reply:
@matthewoliver Yes. the same thing. Error message: "['minion d52-54-77-77-01-05.virtual.cloud.suse.de missing address on public network 192.168.122.0/24', 'minion d52-54-77-77-01-07.virtual.cloud.suse.de missing address on public network 192.168.122.0/24', 'minion d52-54-77-77-01-06.virtual.cloud.suse.de missing address on public network 192.168.122.0/24']"

cluster.yml contains:
public_network: 192.168.122.0/24

But it should be: 192.168.124.0/24

i will try to take a look at where are we ovveriding the net_public env variable.

Reply:
@matthewoliver Perhaps the nodes actually need and admin network? Because they already have it?
So it is just a name collision what we call public is not a public in ses/ceph/rados nomenclature?

After consulting with @bmwiedemann there are 2 suggestions:
1. Do not populate cluster.yml ourselves and let deepsea figure it out by using the default .124.0/24 network.
2. Populate cluster.yml with our admin network address, because that what it expects anyway.